velox/functions/prestosql/tests/ArrayContainsTest.cpp

```diff
@@ -316,4 +317,61 @@ TEST_F(ArrayContainsTest, dictionaryEncodingElements) {
   testContainsConstantKey(arrayVector, {3, 4, 5}, {true, true});
 }
 
+TEST_F(ArrayContainsTest, arrayCheckNulls) {
+  const auto baseVector = makeArrayVectorFromJson<int32_t>({
+      "[1, 1]",
+      "[2, 2]",
+      "[3, null]",
+      "[4, 4]",
+      "[5, 5]",
+      "[6, 6]",
+  });
+  auto data = makeArrayVector({0, 3}, baseVector);
+  auto searchBase =
+      makeArrayVectorFromJson<int32_t>({"[7, null]", "[3, 3]", "[6, null]"});
+  auto search1 = BaseVector::wrapInConstant(data->size(), 0, searchBase);
+  auto search2 = BaseVector::wrapInConstant(data->size(), 1, searchBase);
+  auto search3 = BaseVector::wrapInConstant(data->size(), 2, searchBase);
+
+  static const std::string errorMessage =
+      "contains does not support arrays with elements that are null or contain null";
+  // No null equal.
+  evaluate("contains(c0, c1)", makeRowVector({data, search1}));
+  // Null equal, [3, null] vs [3, 3].
+  VELOX_ASSERT_THROW(
+      evaluate("contains(c0, c1)", makeRowVector({data, search2})),
+      errorMessage);
+  // Null equal, [6, 6] vs [6, null].
+  VELOX_ASSERT_THROW(
+      evaluate("contains(c0, c1)", makeRowVector({data, search3})),
+      errorMessage);
+}
+
+TEST_F(ArrayContainsTest, rowCheckNulls) {
+  const auto baseVector = makeRowVector({
+      makeNullableFlatVector<int32_t>({1, 2, 3, 4, 5, 6}),
+      makeNullableFlatVector<int32_t>({1, 2, std::nullopt, 4, 5, 6}),
+  });
+  auto data = makeArrayVector({0, 3}, baseVector);
+  auto searchBase = makeRowVector({
+      makeFlatVector<int32_t>({7, 3, 6}),
+      makeNullableFlatVector<int32_t>({std::nullopt, 3, std::nullopt}),
+  });
+
+  static const std::string errorMessage =
+      "contains does not support arrays with elements that are null or contain null";
+  auto search1 = BaseVector::wrapInConstant(data->size(), 0, searchBase);
```

Comment:
use makeConstant helper method from the base class

Reply:
Looks like `makeConstant`, `makeConstantArray`, and `makeConstantRow` cannot construct `ArrayOfArray` or `ArrayOfRow with nested null`.